packages/wallets/src/components/Base/WalletClipboard/__tests__/WalletClipboard.spec.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { useCopyToClipboard, useHover } from 'usehooks-ts';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import useDevice from '../../../../hooks/useDevice';
+import WalletClipboard from '../WalletClipboard';
+
+jest.mock('usehooks-ts', () => ({
+    useCopyToClipboard: jest.fn(),
+    useHover: jest.fn(),
+}));
+
+jest.mock('../../../../hooks/useDevice', () => ({
+    __esModule: true,
+    default: jest.fn(),
+}));
+
+describe('WalletClipboard', () => {
+    let mockCopy: jest.Mock;
+    const mockUseDevice = useDevice as jest.Mock;
+
+    beforeEach(() => {
+        mockCopy = jest.fn();
+        (useCopyToClipboard as jest.Mock).mockReturnValue([null, mockCopy]);
+        (useHover as jest.Mock).mockReturnValue(false);
+        mockUseDevice.mockReturnValue({ isMobile: false });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should render the button with copy icon initially', () => {
+        render(<WalletClipboard textCopy='Sample text to copy' />);
+        const button = screen.getByRole('button');
```

Comment:
Nitpick:
This potentially could be a failed spec b/c you don't do 'await' after rendering a component.
_(Especially on React V18 this commonly causes errors b/c of new concurrent rendering feature)_

Potential solution is to always use findBy* (b/c it's async func) or if you're making 'expect' after render or some user event you could wrap it into 'waitFor' func